### PR TITLE
Client list search bar

### PR DESCRIPTION
**Pull request by chaitali-mane**
Signed-off-by: chaitali-mane <cmane@msystechnologies.com>

### :nut_and_bolt: Description: What code changed, and why?
we need to add search bar on client list page.

### :chains: Related Resources
[#4669](https://github.com/chef/automate/issues/4669)

### :+1: Definition of Done
I have added the search bar for Clients list.

### :athletic_shoe: How to Build and Test the Change
if you have sample data for the infra servers than
go to >> infrastructure tab >> click on Chef server on left side navigation bar >> Organizations >> Clients tab >> see the list of Clients.

### :white_check_mark: Checklist

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [ ] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
Client search
clients-list

Client search Value (aws)
Client Search

Client no data in search
client search-no data2

### CODE REVIEW

File: components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        type="text"
+        class="search-input"
+        placeholder="Search {{ placeHolder }}"
+        autocomplete="off">
+      <button chef-button class="search-button primary" primary (click)="handleFiltersClick(search_box.value)">
+        <chef-icon>search</chef-icon>
+      </button>        
+    </div>
+    <ng-content></ng-content>
+  </div>
```
Comment (himanshi-chhabra) [inline components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.html:17]: Add new line at the EOF

File: components/automate-ui/src/app/entities/clients/client.model.ts [hunk 2]
```diff
@@ -17,3 +17,11 @@ export interface ClientKey {
   expiration_date: string;
   private_key: string;
 }
+
+export interface ClientSearch {
```
Comment (himanshi-chhabra) [inline components/automate-ui/src/app/entities/clients/client.model.ts:21]: Remove unused interface.

File: components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.html (hunk 1 above)
Comment (chaitali-mane) [reply components/automate-ui/src/app/modules/infra-proxy/infra-search-bar/infra-search-bar.component.html]: Added new line at EOF

File: components/automate-ui/src/app/entities/clients/client.model.ts (hunk 2 above)
Comment (chaitali-mane) [reply components/automate-ui/src/app/entities/clients/client.model.ts]: Removed interface

**Review by rickmarry — APPROVED** [review]
I'm assuming that this is the same change that pr https://github.com/chef/automate/pull/4693 addressed.. is this just a squashed rebased version of it?

looks good though thanks

**Comment by chaitali-mane** [comment]
In testing found minor issues for header spacing, outline coming for pagination.

**Comment by chaitali-mane** [comment]
Created issue for the UI chages #4750 and raised PR with changes #4754 against issue.

**Pull request merged**